transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -169,50 +171,70 @@ public NetworkEmulator networkEmulator() {
         });
   }
 
-  private Mono<Channel> getOrConnect(Address address) {
+  private Mono<Connection> getOrConnect(Address address) {
+
     return Mono.create(
         sink ->
-            outgoingChannels
+            outgoingConnections
                 .computeIfAbsent(address, this::connect0)
                 .subscribe(sink::success, sink::error));
   }
 
-  private Mono<Channel> connect0(Address address) {
-    return connect1(address)
-        .doOnSuccess(
-            channel -> {
-              LOGGER.debug(
-                  "Connected from {} to {}: {}", TransportImpl.this.address, address, channel);
-
-              ChannelFutureListener closeFutureListener =
-                  future -> {
-                    LOGGER.debug("Disconnected from: {} {}", address, future.channel());
-                    outgoingChannels.remove(address);
-                  };
-
-              channel.closeFuture().addListener(closeFutureListener);
+  private Mono<? extends Connection> connect0(Address address) {
+    return TcpClient.create(ConnectionProvider.fixed("client-" + address, 1))
+        .runOn(loopResources)
+        .host(address.host())
+        .port(address.port())
+        .bootstrap(b -> BootstrapHandlers.updateConfiguration(b, "inbound", outcomingPipeline))
+        .doOnDisconnected(
+            c -> {
+              LOGGER.debug("Disconnected from: {} {}", address, c.channel());
+              outgoingConnections.remove(address);
+            })
+        .doOnConnected(
+            c ->
+                LOGGER.debug(
+                    "Connected from {} to {}: {}",
+                    TransportImpl.this.address,
+                    address,
+                    c.channel()))
+        .connect()
```

Comment:
This `.connect()` should neccessarly go after `.doOnConnected()` and `.doOnDisconnected()`   or  `.connect()`  can be put above those `.doOn...`, `.doOn...` . Just figuring out for myself.

Reply:
just registering some callbacks before emiting signals, not wise-versa